addons/html_editor/static/tests/link/popover.test.js

```diff
@@ -35,6 +35,25 @@ describe("should open a popover", () => {
         setContent(el, "<p>this []is a <a>link</a></p>");
         await expectElementCount(".o-we-linkpopover", 0);
     });
+    test("should close a popover and remove link when click outside of a link without href (2)", async () => {
+        const { el } = await setupEditor("<p>[Hello]</p>");
+        await waitFor(".o-we-toolbar");
+        await click(".o-we-toolbar .fa-link");
+        await waitFor(".o-we-linkpopover", { timeout: 1500 });
+        expect(getContent(el)).toBe(
+            '<p>\ufeff<a class="o_link_in_selection">\ufeffHello\ufeff</a>\ufeff</p>'
+        );
+    });
+    test("should close a popover and remove empty link when press esc", async () => {
+        const { el } = await setupEditor("<p>[Hello]</p>");
+        await waitFor(".o-we-toolbar");
+        await click(".o-we-toolbar .fa-link");
+        await waitFor(".o-we-linkpopover", { timeout: 1500 });
+        await press("escape");
```

Comment:
I would actually "merge" the previous test with this one by checking that, at this point, the link is in the DOM. It makes the test more interesting because at least you would check that something is happening. Right now you start with something, you never check that the DOM ever changes to something else, and then at the end you check that the DOM has correctly reverted back to its original state but we have never seen the intermediate state in between... ^^'

Reply:
Thankss, indeed it's a bit of middle state we check here so I will merge them :pray: